src/Me/Routes/Home/Avatar/Avatar.js

```diff
@@ -0,0 +1,104 @@
+import React, { useContext, useState, useEffect } from 'react';
+import styled from 'styled-components';
+import UserContext from '../../../../context/userContext/UserContext';
+import Camera from '../../../../assets/me/camera.svg';
+import { updateMentorAvatar, getCurrentUser } from '../../../../api';
+import CardContainer from '../../../components/Card/index'
+import { getAvatarUrl } from '../../../../helpers/avatar';
+
+function Avatar() {
+  let { currentUser, updateUser } = useContext(UserContext);
+  const [image, setImage] = useState({ preview: '', raw: '' })
+
+  useEffect(() => {
+    async function initialize() {
+      const user = await getCurrentUser();
+      updateUser(user);
+    }
+    initialize();
+  });
+
+  const handleChange = (e) => {
+    if (e) {
+      setImage({
+        preview: URL.createObjectURL(e.target.files[0]),
+        raw: e.target.files[0]
+      })
+
+      handleUpload(e);
+    }
+  }
+
+  const handleUpload = async (e) => {
+    e.preventDefault()
+    const formData = new FormData()
+    formData.append('image', image.raw)
```

Comment:
This is the reason that sometimes the upload is failed. `setImage` is not changing `image` right away ([more info](https://stackoverflow.com/a/54069332/863110)) so `image.raw` could be empty

I'm not fully understand what we need 2 functions here. Can it be merged to one function?

```js
const handleChange = async (e) => {
  if (e.target.files.length) {
    const formData = new FormData()
    formData.append('image', e.target.files[0])

    const updatedUser = await updateMentorAvatar(currentUser, formData);
    updateUser(updatedUser);
  }
}
```

Make sense?

Reply:
Perfectly, Sorry making it hard on you.
I suppose everything works as expected now, if not just give me a ping.

Reply:
Perfectly, Sorry for making it hard on you.
I suppose everything works as expected now, if not just give me a ping.